packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,589 @@
+/// [SlotLayout].
+///  * [AdaptiveScaffold], which provides a more friendly API with less
+/// customizability. and holds a preset of animations and helper builders.
+///  * [Design Doc](https://flutter.dev/go/adaptive-layout-foldables).
+///  * [Material Design 3 Specifications] (https://m3.material.io/foundations/adaptive-design/overview).
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.topNavigation,
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.bodyOrientation = Axis.horizontal,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the app window; meaning on the left
+  /// when the textDirection is LTR and on the right when it is RTL.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size (width and height) by another method. See the
+  /// builder in [AdaptiveScaffold.toNavigationRail] for an example.
+  final SlotLayout? primaryNavigation;
+
+  /// The slot placed on the end side of the app window; meaning on the right
+  /// when the textDirection is LTR and on the left when it is RTL.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size (width and height) by another method. See the
+  /// builder in [AdaptiveScaffold.toNavigationRail] for an example.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the app window.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size (width and height) by another method. See the
+  /// builder in [AdaptiveScaffold.toNavigationRail] for an example.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the app window.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size (width and height) by another method. See the
+  /// builder in [AdaptiveScaffold.toNavigationRail] for an example.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the app window.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
+  ///
+  /// For example 0.3 would mean [body] takes up 30% of the available space
+  /// and[secondaryBody] takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
```

Comment:
This looks like a weird behavior: If `bodyRatio` is not null, `body` and `secondaryBody` are layouted by _their width_, while if `bodyRatio` is null, `body` and `secondaryBody` are layouted by _the absolute position of the split_. Besides confusion, there is no way to specify a split by the absolute position of the screen other than 50%. What if the fold is not at the center? And also `AdaptiveScaffold` should automatically use the folding information according to `MediaQuery`, right?

I can not think of a good API either. Because we have the following kinds of strategies:
- Automatic (use MediaQuery)
- By the position of the split
- By the width ratio
- By a fixed width of `body`

If I were to suggest, I think this is an API to which we don't know the demand. I would not expose it, and just say that we either use `MediaQuery`'s folding information or we fall back to 50% screen position, and add it later when receive issues.

Reply:
Yes when the bodyRatio is null it uses the MediaQuery to define the split axis along the hinge. I think it is worth exposing it since there are many cases where the main view is smaller than the detail view like in the chat app example in the M3 spec under canonical layouts. I'll change the documentation to clarify what you were saying but if you still think it would be better to not have a bodyRatio then I could remove it.

Reply:
I see. I wonder if the M3 layout is using a ratio or a fixed width. Anyway, let's keep it this way for now.